node/node.cpp

```diff
@@ -117,88 +117,80 @@ node::make_tcp_endpoint(const std::string& host, uint16_t port)
 }
 
 void
-node::connect(const completion_handler_t& callback)
+node::connect()
 {
-    this->strand->post([callback, weak_this = weak_from_this()]()
+    assert(this->state != connect_state::connecting);
+    assert(this->state != connect_state::connected);
```

Comment:
Wouldn't handling this instead of an assert here be more useful?
How can we get into this state? 
Should we do something in release builds? Perhaps a comment is enough to explain this.

Reply:
Should we not try to connect again?

Reply:
We should never actually hit this condition. I put it in there because initially we would try connecting when we were already connecting. I can remove the asserts if you like since they don't really protect any specific variable or crash case.

In terms of connecting again, if the connect fails we will retry the next time someone does a send.

Reply:
It's up to you if you want to remove them.